### PR TITLE
pthread, sysdeps/managarm: add set/get affinity calls

### PR DESCRIPTION
**Pull request by ElectrodeYT**
Depends on [managarm#574](https://github.com/managarm/managarm/pull/574).

### CODE REVIEW

**Review by no92 — APPROVED** [review]
LGTM

File: sysdeps/managarm/generic/sched.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			helix_ng::offer(
+				helix_ng::sendBragiHeadOnly(req, getSysdepsAllocator()),
+				helix_ng::recvInline(),
+				helix_ng::recvBuffer(mask, cpusetsize)
+			)
+		);
+
+	HEL_CHECK(offer.error());
+	HEL_CHECK(send_head.error());
+	HEL_CHECK(recv_resp.error());
```
Comment (qookei) [inline sysdeps/managarm/generic/sched.cpp:38]: Missing check for the `recv_data` item. Also you're not setting `req.size` which the other side expects.

Comment (ElectrodeYT) [reply sysdeps/managarm/generic/sched.cpp:39]: There is a check for `recv_data` on line 50.

Comment (qookei) [reply sysdeps/managarm/generic/sched.cpp:39]: Oh oops.

File: options/posix/generic/pthread-stubs.cpp [hunk 2]
```diff
@@ -309,14 +309,14 @@ int pthread_getattr_np(pthread_t thread, pthread_attr_t *attr) {
...
+	MLIBC_CHECK_OR_ENOSYS(mlibc::sys_getthreadaffinity, -1);
+	return mlibc::sys_getthreadaffinity(reinterpret_cast<Tcb*>(thread)->tid, cpusetsize, mask);
 }
 
-int pthread_setaffinity_np(pthread_t, size_t, const cpu_set_t *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int pthread_setaffinity_np(pthread_t thread, size_t cpusetsize, const cpu_set_t *mask) {
+	MLIBC_CHECK_OR_ENOSYS(mlibc::sys_setthreadaffinity, -1);
+	return mlibc::sys_setthreadaffinity(reinterpret_cast<Tcb*>(thread)->tid, cpusetsize, mask);
```
Comment (qookei) [inline options/posix/generic/pthread-stubs.cpp:319]:
```
RETURN VALUE
       On success, these functions return 0; on error, they return a nonzero error number.
```
These are supposed to return `ENOSYS`, not -1.

Comment (ElectrodeYT) [reply options/posix/generic/pthread-stubs.cpp]: Fixed.

**Review by qookei — APPROVED** [review]

**Pull request merged**